MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+    /// The amount is burnt from the initiator and the module-level nonce is incremented  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    public entry fun initiate_bridge_transfer(  
+        initiator: &signer,  
+        recipient: vector<u8>,  
+        amount: u64  
+    ) {  
+        let initiator_address = signer::address_of(initiator);  
+        let ethereum_address = ethereum::ethereum_address(recipient);  
+    
+        // Increment and retrieve the nonce  
+        let nonce = increment_and_get_nonce(initiator_address);  
+    
+        // Create bridge transfer details  
+        let details = atomic_bridge_store::create_details(  
+            initiator_address,  
+            ethereum_address, 
+            amount,  
+            nonce  
+        );  
+    
+        // Generate a unique bridge transfer ID  
+        let bridge_transfer_id = bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        atomic_bridge_store::add(bridge_transfer_id, details);  
+    
+        // Burn the amount from the initiator  
+        atomic_bridge::burn(initiator_address, amount);  
+    
+        // Emit an event with nonce  
+        event::emit(  
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }  
+
+    /// Completes a bridge transfer by the initiator.  
+    ///  
+    /// @param caller The signer representing the bridge operator.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.  
+    
+    /// @abort If the caller is not the bridge operator.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) {  
+        atomic_bridge_configuration::assert_is_caller_operator(caller);  
+        let ethereum_address = ethereum::ethereum_address(initiator);
+        let details = atomic_bridge_store::create_details(  
+            ethereum_address,  
+            recipient,  
+            amount,  
+            nonce  
+        );
+
+        // asserts that parameters are correct  
+        assert!(bridge_transfer_id == bridge_transfer_id(&details), 0x1);
+        atomic_bridge_store::add(bridge_transfer_id, details);
```

Comment:
This line was supposed to `prepare` the next call (by the user) to complete the bridge transfer.
This part of the algorithm should probably set the status of the transferId to "completed" rather than adding it to the bridge store. 

There should a careful review of what `create_details` and `add` do in the `atomic_bridge_store` to make sure the status of a transfer is updated correctly.